src/python/pants/backend/shell/target_types.py

```diff
@@ -529,6 +536,75 @@ class ShellCommandTestTarget(Target):
     )
 
 
+class ShellCommandPackageDependenciesField(ShellCommandExecutionDependenciesField):
+    pass
+
+
+class ShellCommandPackagedArtifactsField(StringSequenceField):
```

Comment:
DX question: Should the user have to explicitly define what the expected artifacts are for purposes of generating the `BuildPackageArtifact` list?

Reply:
cc @benjyw

Reply:
I think so. The alternative would be what? Take all the output files?

Reply:
The alternative I've seen in the source is to just take the entire output path as a single `BuiltPackageArtifact`. I added this approach in the latest commit if there are no `packaged_artifacts` set. 

To my knowledge, the `BuiltPackageArtifact` instances are not something seen by users and is only really seen by plugin authors who might use package goal output programmatically. Thus, I expect most users are unlikely to care.

I wonder if the "entire output path" artifact should be the only behavior. It would certainly simplify the PR.